File: app/src/main/java/shaiytan/dssanavigator/CarRecord.java
package shaiytan.dssanavigator;

import java.io.Serializable;

/**
 * Created by Shaiytan on 05.05.2017.
 */

public class CarRecord implements Serializable
{

    public String getCarName(){return carName;}
    public void setCarName(String carName){this.carName = carName;}
    public double getLat(){return lat;}
    public void setLat(double lat){this.lat = lat;}
    public double getLon(){return lon;}
    public void setLon(double lon){this.lon = lon;}
    public long getId(){return id;}
    private String carName;
    private double lat;
    private double lon;
    private long id;
    public CarRecord(String carName, double lat, double lon,long id)
    {
        this.carName = carName;
        this.lat = lat;
        this.lon = lon;
        this.id = id;
    }
}

File: app/src/main/java/shaiytan/dssanavigator/model/CarRecord.kt
package shaiytan.dssanavigator.model

import android.content.Context
import android.database.sqlite.*
import shaiytan.dssanavigator.R
import java.io.Serializable


/**
 * Created by Shaiytan on 12.09.2017.
 * Data classes
 */

//data class CarRecord()
data class CarRecord(var carName: String, var lat: Double, var lon: Double, val id: Long) : Serializable

class DBHelper(context: Context) : SQLiteOpenHelper(context, DBNAME, null, 1) {
    companion object {
        private val DBNAME: String = "carstore.db"
    }

    private val res = context.resources
    override fun onCreate(db: SQLiteDatabase) {
        db.execSQL(res.getString(R.string.createQuery))
        db.execSQL("INSERT INTO cars (name, latitude, longitude) VALUES ('TEST', 14, 88);")
    }

    override fun onUpgrade(db: SQLiteDatabase, oldVersion: Int, newVersion: Int) {
        db.execSQL(res.getString(R.string.dropQouery))
        onCreate(db)
    }
}